sharktank/tests/models/llama/benchmark_tests.py

```diff
@@ -0,0 +1,779 @@
+        self.iree_run_prefill_args = [
+            "--function=prefill_bs4",
+            f"--input=@{self.prefill_args_f16}/tokens.npy",
+            f"--input=@{self.prefill_args_f16}/seq_lens.npy",
+            f"--input=@{self.prefill_args_f16}/seq_block_ids.npy",
+            f"--input=@{self.prefill_args_f16}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+        self.iree_run_decode_args = [
+            "--function=decode_bs4",
+            f"--input=@{self.decode_args_f16}/tokens.npy",
+            f"--input=@{self.decode_args_f16}/seq_lens.npy",
+            f"--input=@{self.decode_args_f16}/start_positions.npy",
+            f"--input=@{self.decode_args_f16}/seq_block_ids.npy",
+            f"--input=@{self.decode_args_f16}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+        self.iree_run_prefill_args_fp8 = [
+            "--function=prefill_bs4",
+            f"--input=@{self.prefill_args_fp8}/tokens.npy",
+            f"--input=@{self.prefill_args_fp8}/seq_lens.npy",
+            f"--input=@{self.prefill_args_fp8}/seq_block_ids.npy",
+            f"--input=@{self.prefill_args_fp8}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+        self.iree_run_decode_args_fp8 = [
+            "--function=decode_bs4",
+            f"--input=@{self.decode_args_fp8}/tokens.npy",
+            f"--input=@{self.decode_args_fp8}/seq_lens.npy",
+            f"--input=@{self.decode_args_fp8}/start_positions.npy",
+            f"--input=@{self.decode_args_fp8}/seq_block_ids.npy",
+            f"--input=@{self.decode_args_fp8}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+
+    @longrun
+    def testBenchmark8B_f16_Decomposed(self):
+        output_mlir = self.repo_root + "llama8b_f16_decomposed.mlir"
+        output_json = self.repo_root + "llama8b_f16_decomposed.json"
+        output_vmfb = self.repo_root + "llama8b_f16_decomposed.vmfb"
+        self.export_mlir(
+            "decomposed",
+            self.tensor_parallelism_size,
+            self.irpa_path,
+            output_mlir,
+            output_json,
+            self.repo_root,
+        )
+        iree_compile_args = self.iree_compile_args + [
+            f"--iree-hal-dump-executable-files-to={self.repo_root}/files/llama-8b/f16-decomposed"
+        ]
+        self.iree_compile(output_mlir, output_vmfb, iree_compile_args, self.repo_root)
+        # benchmark prefill
+        self.iree_benchmark_module(
+            "0",
+            output_vmfb,
+            self.irpa_path,
+            self.iree_run_prefill_args,
+            self.repo_root,
+        )
+        # benchmark decode
+        self.iree_benchmark_module(
+            "0",
+            output_vmfb,
+            self.irpa_path,
+            self.iree_run_decode_args,
+            self.repo_root,
+        )
+        self.cleanup_output_files(output_mlir, output_json, output_vmfb)
```

Comment:
To more easily handle regressions and such I think we really want to upload and store the model IR, vmfb, and dispatch dump so we can triage exactly what this benchmark used.

Reply:
Makes sense, I will save the temporary files into the github actions file then.

Reply:
I was kind of thinking of just having it upload to sharkpublic since that would be easy to access for those who want them. Not sure if github actions file would be better or not